### DISCUSSION TITLE
Allow strictExportPresence to work with babel-loader or ts-loader

### DISCUSSION THREAD

**Discussion by skondrashov** [webpack/webpack#18436] [Q&A]
## Feature request

**What is the expected behavior?**
strictExportPresence should check to see if imports are valid in javascript files, and using a loader shouldn't disable this flag.

**What is motivation or use case for adding/changing the behavior?**
It's common for people to use loaders and strictExportPresence is a very helpful flag.

**How should this be implemented in your opinion?**
My understanding is that babel-loader or ts-loader changes files in-place and then passes them on to the webpack parser, which is why `strictExportPresence` is configured under parser/javascript. My understanding is probably wrong but I don't really know why the setting does nothing if you use a loader.

**Are you willing to work on this yourself?**
yes if someone can explain to me why this is happening

**Comment by alexander-akait** [comment]
We validate only pure JS files (after transpile), if you need this for typescript (I think typescript already can do it) and non standard JS syntax (like babel), you need to use linter or ask them to implement this option

**Reply by skondrashov** [reply]
I thought this wouldn't be something a linter would do because linters analyze one file at a time, but I guess there's no reason for that rule to exist. I found https://www.npmjs.com/package/eslint-plugin-import which should do what I want!

There are a lot of javascript files in the project I'm looking at, too many to convert all at once, but we use a loader as we are gradually converting them to typescript. Thank you for your answer! I don't quite understand still why this couldn't be done after transpilation, but it's not particularly important. I guess transpilation by ts-loader or babel-loader already gives you one file? I would have thought it just turns imports into requires or something.
